test/models/supervisor_run_test.rb

```diff
@@ -138,4 +141,26 @@ def setup
     assert_equal @supervisor_script.read_log, "Unable to load log file: #{FILE_PATH}"
   end
 
+  test "proper behaviour of stop method" do
+    @supervisor_script.pid = PID
+    @supervisor_script.is_running = true
+    @supervisor_script.expects(:check).returns(true).twice
```

Comment:
Sam spróbowałem zrobić taki test na szybko i nie za bardzo podoba mi się metoda "check", która oprócz zwracania true/false ustawia flagę is_running, która potem jest sprawdzana w teście. Jeśli chcemy ją zamockować, żeby po wywołaniu TERM zwróciła true, to test i tak nie przejdzie, bo trzeba jeszcze zamockować skutek uboczny (na razie mi się nie udało).

Należałoby albo zaznaczyć, że ta metoda ma skutek uboczny (np. validate_is_running!) i utworzyć drugą funkcję bez skutku uboczengo np. process_running?